### PR TITLE
Update figures with results from extended parameter ranges

### LINKED ISSUE #44: where are the genes going
As more data types are added, we end up with fewer genes (see https://github.com/greenelab/mpmp-manuscript/pull/8). I thought this was because of the inner CV loop ending up with folds with no positive samples, but it looks like sklearn GridSearchCV uses label-stratified CV by default (see https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html).

We should investigate why certain genes are non-deterministically (?) dropping out of certain data types, and think about if there's any way to preserve some of them across analyses.

### PR DESCRIPTION
**Pull request by jjc2718**
Closes #44 (finally - these results should be pretty close to what we want to submit at this point).

The main change in this PR is rerunning all of the single-omics experiments with the correct gene sets and larger hyperparameter ranges (see changes to `mpmp/config.py`), and updating all of the figures with the new results.

The change in hyperparameter ranges caused a fairly substantial change to our results and interpretation: the methylation datasets now perform a lot better, generally on par with gene expression data. We think this is probably because the parameters were optimized at some point in Greg's papers to work well with expression data, so expanding the ranges favored the other data types.

Most of the code changes are related to the figures (either refactoring the plotting code to be a bit simpler, or adding information/details to the figures). Nothing too consequential other than the parameter range updates.

### CODE REVIEW

File: 02_classify_mutations/scripts/run_methylation.sh [hunk 1]
```diff
@@ -1,20 +1,36 @@
...
 N_DIM=5000
 
 mkdir -p $ERRORS_DIR
 
-for data_type in expression me_27k_bmiq; do
-    cmd="python 02_classify_stratified/run_mutation_classification.py --gene_set vogelstein --results_dir $RESULTS_DIR --training_data ${data_type} --seed $SEED 2>$ERRORS_DIR/errors_${data_type}.txt"
-    echo "Running: $cmd"
-    eval $cmd
-done
+for seed in 42 1; do
```
Comment (ben-heil) [inline 02_classify_mutations/scripts/run_methylation.sh:8]: I thought this line was python for a second, and it caused my brain to melt

**Review by ben-heil — APPROVED** [review]
Looks good! Figures 1 and 6 aren't displayed in `generate_figures.ipynb`, though that may be intentional.

**Comment by jjc2718** [comment]
> Looks good! Figures 1 and 6 aren't displayed in `generate_figures.ipynb`, though that may be intentional.

Yep, that's intentional (Figure 1 is generated in BioRender/Inkscape since it's an overview figure/schematic thing, and Figure 6 is just directly generated by another notebook with no panel wrangling required). I'll make this clear in the notebook.
